R/unsupervised.R

```diff
@@ -1,151 +1,188 @@
-.check_tasks <- function(message, tasks, results) {
-  errors <- vapply(results, function (x) inherits(x, "try-error"), logical(1))
-  
-  if (any(errors)) {
-    stop(
-      message,
-      ":\n",
-      paste("\t", tasks[errors], ":", results[errors], collapse = "\n")
-    )
-  }
-  
-  return(results)
+as_feature_crosstab <- function(feature_names, sample_names, data) {
+  colnames(data) <- c('mz', 'rt', 'mz_min', 'mz_max', sample_names)
+  rownames(data) <- feature_names
+  as.data.frame(data)
 }
 
-.check_files <- function(files) {
-  if (any(missing <- !file.exists(files))) {
-    stop(
-      "Cannot find the following files:\n",
-      paste0("\t", files[missing], collapse = "\n")
-    )
+as_feature_sample_table <- function(rt_crosstab, int_crosstab) {
+  feature_names <- rownames(rt_crosstab)
+  sample_names <- colnames(rt_crosstab)[-(1:4)]
+
+  feature_table <- data.frame(
+    feature = feature_names,
+    mz = rt_crosstab[, 1],
+    rt = rt_crosstab[, 2]
+  )
+
+  # series of conversions to produce a table type from data.frame
+  rt_crosstab <- as.table(as.matrix(rt_crosstab[, -(1:4)]))
+  int_crosstab <- as.table(as.matrix(int_crosstab[, -(1:4)]))
+
+  crosstab_axes <- list(feature = feature_names, sample = sample_names)
+  dimnames(rt_crosstab) <- dimnames(int_crosstab) <- crosstab_axes
+
+  x <- as.data.frame(rt_crosstab, responseName = 'sample_rt')
+  y <- as.data.frame(int_crosstab, responseName = 'sample_intensity')
+
+  data <- merge(x, y, by = c('feature', 'sample'))
+  data <- merge(feature_table, data, by = 'feature')
+  data
+}
+
+check_files <- function(filenames) {
+  missing <- !file.exists(filenames)
+  missing_filenames <- paste0('\t', filenames[missing], collapse = '\n')
+
+  if (any(missing)) {
+    stop("Cannot find the following files:\n", missing_filenames)
   }
 }
 
+get_sample_name <- function(filename) {
+  tools::file_path_sans_ext(basename(filename))
+}
+
+sort_samples_by_acquisition_number <- function (filenames) {
+  # assumes that the filenames contain an acquisition number
+  # ideal solution would be to read the acquisition number directly from mzml
+  sort(unlist(filenames))
+}
```

Comment:
Maybe this could also be advanced to take some metadata table which has the injenction order for the batches, same as XCMS and RAMClustR & WaveICA?

Reply:
I am open to suggestions. Do you recommend some text-metafile that will be fed through Galaxy to aplcms? What would it look like?

Reply:
@hechth please create an issue from this comment. (there is also a "reference in new issue" feature on the comment.

Reply:
@martenson Thanks for pointing it out, will do so. Otherwise looks god to me.